Lecture 6 hw

Bepely opened the pull request:
• Добавил правки с **4 лекции** 
• Добавил задание с пятой лекции (_стили для всего проекта_)
• Добавил имитацию логина юзера и переключения темной/светлой темы

Bepely committed:
Lecture 4 Homework

• Added useReducer hook
• Added ReviewForm component
• Added custom scroller

○ Reused counter (for no reason, kinda dumb)

Bepely committed:
Lecture 6 Homework

• Added edits for lecture 4
○Scroller mechanics fixed
○Review mechanics fixed
• Added CSS stylings for lecture 5 HW
• Added Dark/Light theme for lecture 6 HW
• Added user imitation for lecture 6 HW

zubkov7 commented on src/app.jsx:
```diff
@@ -1,31 +1,42 @@
 function App() {
 
     const [currentRestaurant, setCurrentRestaurant] = useState(restaurants[0])
+    const [theme, setTheme] = useState('light')
```
на лекции разбирали подход с косвенным родителем, в текущей реализации будет перерисовываться все приложение при изменении состояния

поэтому работу с состоянием необходимо унести в провайдеры

zubkov7 commented on src/counter/component.jsx:
```diff
@@ -1,11 +1,12 @@
     // Create a counter for added dishes
     const [itemsCounter, setItemsCounter] = useState(0);
 
     const itemsMin = 0;
```
константы лучше выносить из рендер функции, в текущей реализации они будут создаваться на каждый рендер

zubkov7 commented on src/menu/component.jsx:
```diff
@@ -1,13 +1,22 @@
+            <ul >
             {menu.map((element) => {
-                return <li key={element.id}>{element.name} <Counter /></li>
+                return <li className={styles.menuItem} key={element.id}>{element.name} 
```
лучше унести в отдельный компонент меню

zubkov7 commented on src/pageScroller/style.css:
```diff
@@ -0,0 +1,7 @@
+#pageScroller{
```
по айди лучше не применять селекторы, придерживаться классов

zubkov7 commented on src/style.css:
```diff
@@ -0,0 +1,16 @@
+*{
```
лучше не применять стили для всех элементов, так как это может привести к неожиданным поведениям

тут можно сбросить для body например, в остальном лучше в компонентах стилизовать

Bepely committed:
Lecture 6 HW fixes

• States translated to context to avoid re-renders
• Created menuItem component
• Added minor CSS adjustments
